app/modules/auth/auth_router.py

```diff
@@ -36,11 +36,160 @@
 from app.modules.utils.APIRouter import APIRouter
 from app.modules.utils.posthog_helper import PostHogClient
 
+# Import publicsuffix2 for proper domain extraction (handles multi-part TLDs)
+try:
+    from publicsuffix2 import get_sld
+
+    HAS_PUBLICSUFFIX2 = True
+except ImportError:
+    HAS_PUBLICSUFFIX2 = False
+    get_sld = None
+
 SLACK_WEBHOOK_URL = os.getenv("SLACK_WEBHOOK_URL", None)
 
 auth_router = APIRouter()
 load_dotenv(override=True)
 
+# Blocked email domains (generic/personal email providers)
```

Comment:
no library we can use for this instead? if not please move this to the config file